merge_files.py

```diff
@@ -67,16 +78,19 @@ def merge(self):
         end = False
         count = 0
         for i in self.file_list():
-            print(i)
+            logger.info(i)
             with open(i) as input_file:
                 obj = json.load(input_file)
                 self.merge_file(obj, end)
             count += 1
         end = True
         self.merge_file(None, end)
-        print('Done!!')
-        print('{} Files Processed'.format(count))
-        print('{} Result files created'.format(self.counter))
+        the_done_message = 'Done!!'
+        the_message = '{} Files Processed'.format(count)
+        the_files_created = '{} Result files created'.format(self.counter)
+        logger.info(the_done_message)
+        logger.info(the_message)
+        logger.info(the_files_created)

```

Comment:
```suggestion
        logger.info('Done!!')
        logger.info( '{} Files Processed'.format(count))
        logger.info('{} Result files created'.format(self.counter))

```
why use variables here ?
we can directly pass the message to logger.
I am new to this so, could you offer an explanation ?

Reply:
Yes, you are correct. for my first pull request I did this. However, pylinter didnt like that. If you would like I can editi it back to the way you would like it :). please let me know!

Reply:
ah, I see, its the .format() method
more on that [here](https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation#:~:text=pylint%20produces%20the%20following%20warning,%2Dformat%2Dinterpolation%20(W1202)%3A&text=Such%20calls%20should%20use%20%25%20formatting,passing%20the%20parameters%20as%20arguments.)

```
logger.info( '{} Files Processed'.format(count))
logger.info('{} Result files created'.format(self.counter))
```
change the above to
```
logger.info( '%s Files Processed', count)
logger.info('%s Result files created', self.counter) 
```
I think this should work

Reply:
Thank you for letting me know. I did not know this.

Reply:
neither did I, guess we learn something new everyday

Reply:
It reminds me of when I learned C. we used to use that syntax everywhere. I didnt understand why the school taught me it, but now I understand they wanted me to be familiar with different ways the computer can register input/output.

Reply:
It failed the linter again. It seems like the linter liked the first version more.

Reply:
if you click on details under checks you should see the error message on why they are failing.
In this case:
_`Missing function or method docstring (missing-function-docstring)`_
Which suggests you are missing a docstring in the new function format_logger() you defined.
Add a docstring with a short explanation of the function, like this:
```
def format_logger():
    """
    configures the logger object
    """
```
You can change the content of the docstring to what you think, it doesnt have to be the same as I mentioned above

_Also on a side note, the name format_logger doesnt seem intuitive, perhaps we can change it to configure_logger?_

Reply:
> It reminds me of when I learned C. we used to use that syntax everywhere. I didnt understand why the school taught me it, but now I understand they wanted me to be familiar with different ways the computer can register input/output.

its not really needed since we have f-string formatting and .format() method , 
but logging implements it for supporting previous versions, it will be obsolete/optional in logging library too in near future I think